### PR TITLE
Pull Request

### PR DESCRIPTION
**Pull request by MaxMartinsen**
Profile updates do not currently work on the personal profile page, but all of them are displayed on the feed page. At the moment I am working on the transition by id as well as deletion and edit. Please give me your feedback.

### CODE REVIEW

**Review by sulenchy — CHANGES_REQUESTED** [review]
@MaxMartinsen You have 2065 files change? That is too much. You need to gitignore node_modules and dist folder.

**Comment by MaxMartinsen** [comment]
Please look at my work and leave a review.
Now working on the profile page so that the feed of posts only from the profile is visible.
The post feed requires your attention, perhaps checking the URL of the text or image is not enough?
I'm not happy with the names of some of the functions; going to redesign them and make them more modular, I think this will help make the code more readable.

**Comment by MaxMartinsen** [comment]
Remove node_modules from version control

File: src/js/auth/login.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+/**
+ * Login a user by making a POST request to the provided URL with the given user data.
```
Comment (sulenchy) [inline src/js/auth/login.js]: This looks good to me

File: src/js/auth/register.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+/**
```
Comment (sulenchy) [inline src/js/auth/register.js]: This looks exactly like the login functionality. Why can't you just stick to one and remove the other? You can rename the loginUser function to a common name such as `fetch_auth`

File: src/js/components/formatDateAndTime.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+export const formatDateAndTime = (dateString) => {
```
Comment (sulenchy) [inline src/js/components/formatDateAndTime.js]: This file looks like a utility function and not a component. A component is more of UI than anything and should have DOM manipulation kind of code. Why don't you move it out to another folder such as `utils`.

File: src/js/components/formatTag.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+export function formatTags(tags) {
```
Comment (sulenchy) [inline src/js/components/formatTag.js]: Same thing as above

File: src/js/components/setActiveFilter.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+export function setActiveFilter(filterId) {
```
Comment (sulenchy) [inline src/js/components/setActiveFilter.js]: Same as above

File: src/js/display/displayPost.js [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+        const post = await fetchGetPostById(postId);
+
+        // Retrieve the current user's ID
+        const currentUserId = getCurrentUserId();
+
+        // Define necessary variables
+        const postTitle = post.title ? `<h3>${post.title}</h3>` : '';
+        const postAuthor = post.author?.name || "Unknown Author";
+        const userAvatar = post.author?.avatar || "../../images/img/avatar/default-avatar.jpg";
+        const postImage = post.media ? `<img src="${post.media}" alt="Post Image" class="img-fluid rounded mb-3">` : '';
```
Comment (sulenchy) [inline src/js/display/displayPost.js]: Good job! for having a placeholder for post image

File: src/js/display/displayPost.js [hunk 7]
```diff
@@ -0,0 +1,77 @@
+import { fetchGetPostById } from "../request/fetchGetPostById.js";
+import { formatDateAndTime } from "/src/js/components/formatDateAndTime.js";
+import { formatTags } from "/src/js/components/formatTag.js";
+
+
+export async function displayPost() {
+    try {
+        // Extract the post ID from the URL
```
Comment (sulenchy) [inline src/js/display/displayPost.js]: I see that you are repeating in other places. Don’t you think having it as a util function would be better?

**Review by MarvyK — COMMENTED** [review]
@MaxMartinsen  - thanks for your submission of your PR. Good job on your submission. You can go ahead and merge into your main branch. Well done!

**Pull request merged**